stmhal/timer.c

```diff
@@ -1095,9 +1104,12 @@ STATIC mp_obj_t pyb_timer_callback(mp_obj_t self_in, mp_obj_t callback) {
         __HAL_TIM_DISABLE_IT(&self->tim, TIM_IT_UPDATE);
         self->callback = mp_const_none;
     } else if (mp_obj_is_callable(callback)) {
+        __HAL_TIM_DISABLE_IT(&self->tim, TIM_IT_UPDATE);
```

Comment:
Does it really need to be disabled?  And then the flag cleared below?  Wouldn't this potentially miss an IRQ?  Isn't it better to do the minimal stuff here rather than reset everything?